### PR TITLE
Help Center: add the new Help Center popover

### LINKED ISSUE #69617: Task: Help Center: build a promotional popover for the Help Center
### Details

In similar fashion to this PR https://github.com/Automattic/wp-calypso/pull/62605, we need to inform users (exactly once) about the new Help Center. This implies the following:

1. Create a V2 public-api endpoint (not an ETK one), to write and read a [single Calypso preference](fbhepr%2Skers%2Sjcpbz%2Schoyvp.ncv%2Serfg%2Sjcpbz%2Qwfba-raqcbvagf%2Spynff.jcpbz-wfba-ncv-zr-cersreraprf-i1-1-raqcbvag.cuc%3Se%3Qrnr82sq4-og). Call it `dismissible-card-masterbar-collapsable-help-center-popover`.
2. Create an ETK proxy URL, so Atomic users can access this value.
3. Create a `react-query` hook that reads this value and decides whether or now to show the popover. Make sure this hooks stores the info forever (`staleTime: Infinity`). 
4. Create a `useMuation` mutation that changes the value and invalidates the cache from the query above. Like [so](https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/blog-stickers/use-remove-blog-sticker-mutation.js#L23).

### Checklist

_No response_

### Related

_No response_

### PR DESCRIPTION
**Pull request by alshakero**
#### Proposed Changes

* This adds a new popover to the Help Center to promote it like so: 
image
* It works in Calypso and the Editor, but still needs some work on Atomic.

#### Testing Instructions

1. Checkout this branch.
2. Apply this patch D92371-code.
3. Open Calypso my Home. You should see the popover.
4. Go to the Editor, you should see it too. 

If you want to reset the preference value, you can check these instructions: https://github.com/Automattic/wp-calypso/pull/62605#:~:text=you%20like%20here%3A-,To%20see%20the%20popover%20again,-%2C%20please%20untick%20%22dismissed

Fixes https://github.com/Automattic/wp-calypso/issues/69617

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-has-seen-promotion.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	/**
+	 * Callback to determine whether the request can proceed.
+	 *
+	 * @return boolean
+	 */
+	public function permission_callback() {
+		return is_user_logged_in();
+	}
+}
```
Comment (alshakero) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-has-seen-promotion.php]: We shouldn't allow Atomic sites to touch user preferences wholesale, we should either add a whitelist of prefs, or only allow one prop (`seen_help_center_promotion`).

File: packages/help-center/src/components/help-icon.tsx [hunk 2]
```diff
@@ -1,23 +1,85 @@
...
+					setHasSeenPromotionalPopover( true );
+				} }
+				position="bottom left"
+				showDelay={ 500 }
+			>
+				<div className="help-center__promotion-popover-inner">
+					<h1>{ __( '✨ Our new Help Center', __i18n_text_domain__ ) }</h1>
+					<p>
+						{ __(
+							'We moved our help resources! You can always find it int he top bar.',
```
Comment (alshakero) [inline packages/help-center/src/components/help-icon.tsx]: Typo 
```suggestion
							'We moved our help resources! You can always find it in the top bar.',
```

**Review by alshakero — COMMENTED** [review]
Typo.

File: packages/help-center/src/components/help-icon.tsx (hunk 2 above)
Comment (arcangelini) [reply packages/help-center/src/components/help-icon.tsx]: 👍

**Comment by arcangelini** [comment]
Tested in...

- Simple - wp-admin/editor (post,site,widget) ✅ appear ✅  dismiss
- Atomic - wp-admin/editor (post,site,widget) ✅ appear ✅  dismiss
- Simple - calyspo/editor ✅ appear ✅  dismiss

**Review by arcangelini — APPROVED** [review]
Tested collectively in a huddle. Everything is working great!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7633820

Thank you @alshakero for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7633820).

**Pull request merged**